drivers/volume/portworx/portworx.go

```diff
@@ -428,6 +476,29 @@ func (p *portworx) SnapshotCreate(
 		return nil, getErrorSnapshotConditions(err), err
 	}
 
+	if snap.Metadata.Annotations != nil {
```

Comment:
This logic shouldn't be in the portworx driver. It is generic and needs to be in a common place.

Reply:
moved all of the rule execution and parsing to their own packages now.

I had to keep atleast the below section here since we need the control here for doing things like terminating the commands early for cloudsnaps the moment they are triggered.